ui/legacy/src/components/cstudio-forms/data-sources/receptacles.js

```diff
@@ -0,0 +1,355 @@
+/*
+ * Copyright (C) 2007-2020 Crafter Software Corporation. All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as published by
+ * the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+(function () {
+
+  function Receptacles(id, form, properties, constraints) {
+    this.id = id;
+    this.form = form;
+    this.properties = properties;
+    this.constraints = constraints;
+    this.type = '';
+    this.allowShared = false;
+    this.allowEmbedded = false;
+    this.enableSearch = false;
+    this.enableBrowse = false;
+    this.baseRepoPath = '/site/components';
+    this.baseBrowsePath = '/site/components';
+    //this.countOptions = 0;
+    const i18n = CrafterCMSNext.i18n;
+    this.formatMessage = i18n.intl.formatMessage;
```

Comment:
Extract this out of the class. Use it as a function instead of a class prop. Same for messages.